### ISSUE TITLE
fengari.io

### ISSUE THREAD

**Issue by Bat1963** [fengari-lua/fengari#216]
Hello, the website (fengari.io) seems to have disappeared. I eventually learnt that it was very useful to test stuff out. Does it need a new VPS or something? I have a few cheappie ones if it will run from a folder. Happy to contribute towards this as I haven't donated anything to the project so far.

**Comment by daurnimator** [comment]
I think @giann got sick of paying the renewal fee for the `.io` which is non-trivial?

I have VPS(s) we can host on. Or really we could just host the site off github pages because it's all static.

**Comment by Bat1963** [comment]
Could you possibly let me know the domain name cost and whether Giann would be willing to reinstate it with some financiaI help. Hopefully not loads. I will be able to let you know if I can help. Fengari.net isn't too dear. Cheers for now. 

-- 
  Jon Cooper
  ***@***.***

**Comment by giann** [comment]
I still have the domain but i shut down the digital ocean droplet.

**Comment by Bat1963** [comment]
Presumably that's the expensive bit? I've never used DO, sorry, just plain simple stuff. Does DO give you a certificate or something? I haven't tangled with https stuff either. How much is the droplet and what does it do ;-? Complete DO numpty here. 

-- 
  Jon Cooper
  ***@***.***

**Comment by Bat1963** [comment]
Apologies, I see a do droplet is a vps. I have 3 cheappie ones (don't we all) and am happy in priinciple to take forwards from the domain, if that is any use. Just need to get my head around the details. But it could happen - just let me know.

-- 
  Jon Cooper
  ***@***.***

**Comment by ryanford** [comment]
Does the website need a backend? Taking a peak on the site repo looks like it's just using lapis to serve etlua templating but that could be compiled AoT and served as static files. I think it be hosted on GitHub pages without cost to you.

**Comment by Bat1963** [comment]
Cool, I only know lighttpd so I am not sure I can help easily but happy to try. Just let me know. 

-- 
  Jon Cooper
  ***@***.***

**Comment by garryspins** [comment]
We really could use this back up

**Comment by Bat1963** [comment]
I agree. Folk have suggestions and I am going to see if I can help although I don't have any experience in what's needed, plus other deadlines (damnation). 

-- 
  Jon Cooper
  ***@***.***

**Comment by giann** [comment]
An easier option might be to donate enough through github sponsor for me to pay the VPS.

Otherwise anyone able to setup the site can show it to me and i can point the DNS to it.

**Comment by Bat1963** [comment]
Donated ;-)

**Comment by leafo** [comment]
I deployed it here: https://fengari.moonscript.org/

Currently cloudflare to proxy for https. If you want to point to there (CNAME to fengari.moonscript.org) and don't want to use something like cloudflare to enable https then I can set up letsencrypt for the domain after it's pointing there.

**Comment by giann** [comment]
Thanks @leafo. I'll update the DNS tomorrow.

@Bat1963 let me know if you want your donation refunded.

**Comment by Bat1963** [comment]
No worries, my friend, you funded this enough yourself over the years, so I am happy to contribute. 

I recently added a bit to my tutorial here: https://www.ucl.ac.uk/~rmhajc0/fengarilua.html

If you have a minute, please let me know of any major bloomers or omissions. 

Best wishes, Jon.C.

**Comment by vitiral** [comment]
that's what you (hopefully) get when it's 100% self-contained html and javascript 😄

**Comment by Bat1963** [comment]
For a laugh, I saved the files with my browser, put them here and it works: http://ic50.org/fengari/ but its not https.

**Comment by giann** [comment]
@leafo can you give me an IP? A redirection with CNAME is only possible for a subdomain.

**Comment by giann** [comment]
> I recently added a bit to my tutorial here: https://www.ucl.ac.uk/~rmhajc0/fengarilua.html

@Bat1963 This is cool! Maybe format the code. 0 indentation is a little hardcore :)

**Comment by Bat1963** [comment]
This one is a bit prettier: https://justpaste.it/8cb7f

**Comment by leafo** [comment]
> @leafo can you give me an IP? A redirection with `CNAME` is only possible for a subdomain.
> 
> Edit: Trying with an `ALIAS` record just now

You can use 173.255.218.206

**Comment by giann** [comment]
@leafo fengari.io point to that IP now but https is ko.

**Comment by Bat1963** [comment]
Great work. Thank you both so much for doing this. Amazingly, I did manage to get letsencrypt, certbot and lighttpd to all work on one of my VPS's but I haven't done it for the one mentioned above yet. Do let me know if that would be of any use. 
-- 
  Jon Cooper
  ***@***.***

**Comment by Bat1963** [comment]
P.S. Also, it would be great to have the web site mentioned in the side bar on github again. 

-- 
  Jon Cooper
  ***@***.***

**Comment by daurnimator** [comment]
> that's what you (hopefully) get when it's 100% self-contained html and javascript 😄

*html, javascript and lua

:)


> Lapis can be used to generate a static version of the site

oh interesting!

But looks like it's all being handled now

**Comment by Bat1963** [comment]
Any chance we can put a link to fengari.io back in the github pages? I think it would help as it flags up what fengari can do. Just trying to sell this stuff for you folks ;-0 ;-0

**Comment by giann** [comment]
@Bat1963 I'd like https to be fixed first. @leafo can you look into it?

**Comment by Bat1963** [comment]
Hello

After a long battle with openvz running an old version of ubuntu, lighttpd and certbot, this one is https now. 

https://ic50.org/fengari/

Feel free to forward to it. I try to maintain that site quite regularly. 

Best wishes
-- 
  Jon Cooper
  ***@***.***

**Comment by Bat1963** [comment]
Hello folks. Well, there's now a version of the fengari.io website that is https of sorts (my previous msg). I think it's OK. Just wondering if that would help with putting the website back on the github page? Best wishes from Bat1963

**Comment by cattokomo** [comment]
There should be a Github pages hosted of fengari.io if the community-provided one are all dead.

**Comment by Bat1963** [comment]
Hello, I put the site here on github and it seems to work OK: https://jonbcooper.github.io/fengari/

It took me about a hour to cobble it together from virtually zero github knowledge by googling.

**Comment by daurnimator** [comment]
@leafo TLS seems to not be working? Is that something easy to fix?

**Comment by wrq** [comment]
Sorry to resurrect this thread, but fengari.io currently displays a DNS resolution error. Perhaps a misconfigured Cloudflare setup?

**Comment by Bat1963** [comment]
Yeah, I wish the fengari folks would sort it out. It's a great program. Is it a question of buying a certificate or something? You can get round that with certbot, etc, but it can be fiddly. 

-- 
  Jon Cooper
  ***@***.***

**Comment by leafo** [comment]
The site is also running at https://fengari.moonscript.org/

http://fengari.io works but not https. I can't set up certbot to get a certificate for the domain because Cloudflare appears to be routing it and it's blocking it via some bot mode from what I can tell. I don't own the domain, just the server where the DNS is pointing at.
